data/sql_setup/prepare_schedule_b.sql

```diff
@@ -18,49 +18,86 @@ create index on ofec_sched_b_queue_old (timestamp);
 create index on ofec_sched_b_queue_new (two_year_transaction_period);
 create index on ofec_sched_b_queue_old (two_year_transaction_period);
 
--- Create trigger to maintain Schedule B queues
-create or replace function ofec_sched_b_update_queues() returns trigger as $$
+-- Create trigger to maintain Schedule A queues for inserts and updates
+-- These happen after a row is inserted/updated so that we can leverage pulling
+-- the new record information from the view itself, which contains the data in
+-- the structure that our tables expect it to be in.
+create or replace function ofec_sched_b_insert_update_queues() returns trigger as $$
 declare
     start_year int = TG_ARGV[0]::int;
     timestamp timestamp = current_timestamp;
     two_year_transaction_period_new smallint;
     two_year_transaction_period_old smallint;
+    view_row fec_vsum_sched_b_vw%ROWTYPE;
 begin
     if tg_op = 'INSERT' then
-        two_year_transaction_period_new = get_transaction_year(new.disb_dt, new.rpt_yr);
+        select into view_row * from fec_vsum_sched_b_vw where sub_id = new.sub_id;
+        two_year_transaction_period_new = get_transaction_year(new.disb_dt, view_row.rpt_yr);
 
         if two_year_transaction_period_new >= start_year then
-            delete from ofec_sched_b_queue_new where sub_id = new.sub_id;
-            insert into ofec_sched_b_queue_new values (new.*, timestamp, two_year_transaction_period_new);
+            delete from ofec_sched_b_queue_new where sub_id = view_row.sub_id;
+            insert into ofec_sched_b_queue_new values (view_row.*, timestamp, two_year_transaction_period_new);
```

Comment:
For this file, [line 31](https://github.com/18F/openFEC/pull/2218/files/473c77d571b9a3c4a49461fedc7dc8a19875da64#diff-5cf905b76958badfe6708f3445af7459R31) and [line 34](https://github.com/18F/openFEC/pull/2218/files/473c77d571b9a3c4a49461fedc7dc8a19875da64#diff-5cf905b76958badfe6708f3445af7459R34) are what take care of `view_row`.  Line 31 defines the variable and sets it to be a record of type `fec_vsum_sched_b_vw%ROWTYPE` - this means it carries the same column structure as what is in the view (could be a table or materialized view too).  Without this specific definition (for instance, if I had just gone with `view_row RECORD` instead), PostgreSQL would not know what to do with the data when utilizing it elsewhere, e.g, an insert into somewhere else, because it has no "type" and therefore can't be registered, [which results in an error like this](http://stackoverflow.com/questions/37815054/getting-an-error-record-type-has-not-been-registered-in-postgresql-what-is-wr).  By setting the type explicitly (which is awesome that you can do this), we can use the record more directly and map things very easily. :-)

Line 34 is what actually populates the record into the `view_row` variable.  Within a stored procedure/function like this (at least with PL/pgSQL), you can [select data from the database into a variable](https://www.postgresql.org/docs/current/static/plpgsql-statements.html#PLPGSQL-STATEMENTS-ASSIGNMENT) to be used elsewhere.  Since we want to take that `view_row` with the record to be processed pretty much as is but check a couple of things first, this makes it possible to do all of that.

As for what will actually be placed into the queues, yes, only records that are reflected in the views, and the triggers on the source table are what guarantee that.  None of this would happen without a change to a record in the source table, but as soon as that change is about to happen or just happened, accessing the view would reflect the current state of the data from the source tables.